packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1142,6 +1144,11 @@ export class TransactionController extends BaseControllerV1<
 
   startTrackingByNetworkClientId(networkClientId: NetworkClientId) {
     const networkClient = this.getNetworkClientById(networkClientId);
+    const chainId = networkClient.configuration.chainId;
+    if(!this.nonceMutexByChainId.get(chainId)) {
+      this.nonceMutexByChainId.set(chainId, new Mutex())
+    }
```

Comment:
need to clear mutexes on network removal either keeping count of networks left with chainId, or just checking if any remaining networks with chainId